packages/core/src/electron-browser/menu/electron-main-menu-factory.ts

```diff
@@ -183,22 +176,22 @@ export class ElectronMainMenuFactory extends BrowserMainMenuFactory {
 
             const accelerator = bindings[0] && this.acceleratorFor(bindings[0]);
 
-            const menuItem: Electron.MenuItemConstructorOptions = {
+            const menuItem: MenuDto = {
                 id: node.id,
                 label: node.label,
                 type: this.commandRegistry.getToggledHandler(commandId, ...args) ? 'checkbox' : 'normal',
                 checked: this.commandRegistry.isToggled(commandId, ...args),
                 enabled: true, // https://github.com/eclipse-theia/theia/issues/446
                 visible: true,
                 accelerator,
-                click: () => this.execute(commandId, args, options.rootMenuPath)
+                execute: () => this.execute(commandId, args, options.rootMenuPath)
```

Comment:
How can a "DTO" contain functions?

Reply:
`Menu` is heavily overloaded and I didn't want to introduce another, similarly named entity. I think it kinda makes sense, since the intent of the object is to be transfered to the main process (with some massaging).

Reply:
I had time to get familiar with the code as well as Electron's APIs, and Electron allows functions in objects passed between the browser and preload contexts.

I don't have a better naming for `MenuDto` either so I'm fine with it.